Add editorial for "Left Out" (#6240)
cpinitiative/usaco-guide: solutions/silver/usaco-942.mdx

2. Now, the entire first column should be L.
3. If the first cow in any column is an R, then we turn the entire column.
4. Now, the entire first row should be L.

Now, we must consider the region that is neither in the first row nor the first column.
First, if there's an R in the interior region, then it is impossible to turn the entire grid to L;
attempts to get rid of that R by flipping the row or column it is contained in will result in another R in the first row or column, which means that it is impossible to get rid of all R's in this state.

In that case:
- If there's only one R in the interior region, then that R is the nonconforming cow requested by the problem.
- If there's only a single row or column of R's in that region, then first, we flip that row or column in the overall grid, and the first cow of the overall row or column will be the only R, which makes it the wanted cow.
- If the entire interior region is filled with R's, then we flip every row besides the first row, and then flip the first column in the entire grid; now, only $(1,1)$ will be an R, which makes that our answer.
- The above covers cases where there's a single left-out cow at any location;

>   therefore, for anything else, there is no way to make the grid uniform by just flipping one cow.

Reviewer: these feel like run-on sentences ngl

would prob benefit to change the syntax a bit

Author: ```suggestion
In that case, there are several cases for the locations of R's in the region:
- **If there's only one R:** That R is the nonconforming cow requested by the problem.
- **If there's only a single row or column of R's:** Flip that row or column in the entire grid, and then the first cow in that row or column will be the only R, which gives us the answer.
- **If the entire region is filled with R's:**  First, flip every row besides the first row in the entire grid, and then flip the first column in the entire grid; only $(1,1)$ will be an R, which makes that our answer.
- **Any other pattern of R's:** The above cases covers single left-out cows at any location;
  therefore, in this case, there is no way to make the grid uniform by just flipping one cow.
```

@eysbutno Does this seem better?

---
id: usaco-942
source: USACO Silver 2019 Open
title: Left Out
author: John Zhou
---

[Official Analysis (C++)](https://usaco.org/current/data/sol_leftout_silver_open19.html)

## Explanation

Below, we will turn most of the cows to L, and the cow that is "left out" will be R;

> neither L nor R have a benefit over the other direction, as the final direction can be switched by turning all rows.

Reviewer: ```suggestion
Without loss of generality, we have the left-out cow face right, and the rest facing left. Since we can flip the direction of all the cows, this covers the equivalent case of the left-out cow facing left.
```

Author: I've been debating whether to use `WLOG`—since it normally applies to swapping variables, not an extra operation at the end...  What do you think?

Reviewer: > The term is used to indicate the assumption that what follows is chosen arbitrarily, narrowing the premise to a particular case, but does not affect the validity of the [proof](https://en.wikipedia.org/wiki/Mathematical_proof) in general.

seems pretty in line with how it's used, idk why it would be restricted to swapping variables

Author: Nevermind... the extra operation *is* just swapping variables, so WLOG actually makes it more clear.

Note that turning the same row or column twice has no effect.
Therefore, initially, we can simply consider whether to turn each row and column; first, we consider each row.
If the first cow in a row is an R, then we turn the entire row;
this will result in the entire first column of cows is L.
Next, we consider each column: If the first cow in a column is an R, then we turn the entire column;

> this will ensure that the entire first row of cows is L.

Reviewer: maybe bullet points? that may be more readable IMO

Author: ```suggestion
Note that turning the same row or column twice has no effect.
Therefore, we can simply consider whether to turn each row and column.
We use the following procedure initially:
1. If the first cow in any row is an R, then we turn the entire row.
2. Now, the entire first column should be L.
3. If the first cow in any column is an R, then we turn the entire column.
4. Now, the entire first row should be L.
```

Now, we must consider the region that is not in the first row or column, herein referred to as the "interior region".
If the grid does not contain a cow that is "left out", then the interior reigon should be L at this point:
if there's an R in that region, then we must flip its row or column to get rid of it, but that'll mean that we'd end up with another R in the first row or column, ultimately rendering the puzzle unsolvable.
In that case, there may be a cow that is left out:
- If there is only one R in the interior region, then that R is the left-out cow, as we may turn it to match the others.
- If there is only a single row or column of R's in the interior region, then first, we flip that row or column in the overall grid, and the first cow of the overall row or column will be left out.
- If the entire interior region is filled with R's, then actually, we may flip the first row and first column of the grid; now, the entire grid will be R's, except for $(1, 1)$, which is our left-out cow.

> - The above covers cases where there's a single left-out cow at any location;

Reviewer: IMO the writing quality here can be improved; for example:
> If the grid does not contain a cow that is "left out", then the interior reigon should be L at this point:
if there's an R in that region, then we must flip its row or column to get rid of it, but that'll mean that we'd end up with another R in the first row or column, ultimately rendering the puzzle unsolvable.

doesn't really read too nice

Author: ```suggestion
Now, we must consider the region that is neither in the first row nor the first column.
First, if there's an R in the interior region, then it is impossible to turn the entire grid to L;
attempts to get rid of that R by flipping the row or column it is contained in will result in another R in the first row or column, which means that it is impossible to get rid of all R's in this state.

In that case:
- If there's only one R in the interior region, then that R is the nonconforming cow requested by the problem.
- If there's only a single row or column of R's in that region, then first, we flip that row or column in the overall grid, and the first cow of the overall row or column will be the only R, which makes it the wanted cow.
- If the entire interior region is filled with R's, then we flip every row besides the first row, and then flip the first column in the entire grid; now, only $(1,1)$ will be an R, which makes that our answer.
- The above covers cases where there's a single left-out cow at any location;
```

  therefore, for anything else, there is no way to make the grid uniform by just flipping one cow.

We can check all the above cases one by one, and output the coordinate of the left-out cow appropriately

> (or $-1$ if either no individual turns are needed, or if it is impossible with $1$ individual turn.).

Reviewer: you can probably get rid of this sentence tbh

Author: ```suggestion
```
